packages/dm-core/src/components/ViewCreator/ViewCreator.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  isRecipeViewConfig,
+  isReferenceViewConfig,
+  TViewConfig,
+} from '../../types'
+import { TGenericObject, UiPluginContext } from '../../index'
+import React, { useContext } from 'react'
+import { ReferenceView } from './ReferenceView'
+import { RecipeView } from './RecipeView'
+import { getTarget, getType } from './utils'
+
+type TViewCreator = {
+  idReference: string
+  document: TGenericObject
```

Comment:
Should it be up to the parent, or ViewCreator to fetch the document, at the correct depth.
I think it should be handled by "ViewCreator", to make the component as simple as possible to use.

Reply:
I also think this should be up to the ViewCreator. Right now the ViewCreator only fetches the document to get the list of UI recipes. 
Not sure if we need to fetch it for something else, maybe when we will support dotted paths for scope?

Reply:
I guess it is possible to add depth as an attribute in the ViewConfig blueprint so that it can be used as input to either control the depth inside ViewCreator or passed to the plugin to control the depth for what happens inside the plugin.

Reply:
Added issue for this: https://github.com/equinor/dm-core-packages/issues/133